### PR TITLE
Misc. typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found via `codespell -q 3` in downstream vtk/paraview

### CODE REVIEW

File: src/metaArrow.h [hunk 1]
```diff
@@ -72,8 +72,8 @@ class METAIO_EXPORT MetaArrow : public MetaObject
     void  Length(float length);
     float Length(void) const;
 
-    void  Lenght(float length) {this->Length(length);}
-    float Lenght(void) const {return Length();}
+    void  Length(float length) {this->Length(length);}
+    float Length(void) const {return Length();}
```
Comment (luzpaz) [inline src/metaArrow.h]: please review this specific change.

File: itkzlib/zlib.h [hunk 2]
```diff
@@ -1085,7 +1085,7 @@ ZEXTERN gzFile ZEXPORT gzopen  OF((const char *path, const char *mode));
 ZEXTERN gzFile ZEXPORT gzdopen  OF((int fd, const char *mode));
 /*
      gzdopen() associates a gzFile with the file descriptor fd.  File
-   descriptors are obtained from calls like open, dup, creat, pipe or
+   descriptors are obtained from calls like open, dup, create, pipe or
```
Comment (luzpaz) [inline itkzlib/zlib.h]: please review

File: src/metaCommand.cxx [hunk 3]
```diff
@@ -127,7 +127,7 @@ SetOption(METAIO_STL::string name,
               << " is expecting a shortTag of exactly one character."
               << " You should use the SetOptionLongTag(optionName,longTagName)"
               << " if you want to use a longer tag. The longtag will be"
-              << " refered as --LongTag and the short tag as -ShortTag."
+              << " referred as --LongTag and the short tag as -ShortTag."
```
Comment (luzpaz) [inline src/metaCommand.cxx]: perhaps should be `referred to as` ?

File: src/metaCommand.cxx [hunk 4]
```diff
@@ -174,7 +174,7 @@ SetOption(METAIO_STL::string name,
               << " is expecting a shortTag of exactly one character."
               << " You should use the SetOptionLongTag(optionName,longTagName)"
               << " if you want to use a longer tag. The longtag will be "
-              << " refered as --LongTag and the short tag as -ShortTag "
+              << " referred as --LongTag and the short tag as -ShortTag "
```
Comment (luzpaz) [inline src/metaCommand.cxx]: same as above

File: itkzlib/zlib.h (hunk 2 above)
Comment (thewtex) [reply itkzlib/zlib.h]: This is supposed to be `creat`:

  https://www.zlib.net/manual.html

File: src/metaArrow.h (hunk 1 above)
Comment (thewtex) [reply src/metaArrow.h]: It looks like this was left for backwards compatibility and should be retained.

File: src/metaCommand.cxx (hunk 3 above)
Comment (thewtex) [reply src/metaCommand.cxx]: +1

File: src/metaCommand.cxx (hunk 4 above)
Comment (thewtex) [reply src/metaCommand.cxx]: +1

**Comment by luzpaz** [comment]
made appropriate revisions based on @thewtex review.

**Pull request merged**